common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java

```diff
@@ -327,6 +403,12 @@ private Mono<Void> runSyncOrAsync() {
         return Mono.defer(() -> {
             if (options.isSync()) {
                 return Mono.fromFuture(CompletableFuture.supplyAsync(() -> runTest())).then();
+            } else if (options.isCompletableFuture()) {
```

Comment:
@alzimmermsft do you think we should update this to non-reactor too?

Reply:
It will add some overhead, but I don't know if it'll be enough to drastically change results.

(Side comment, I think overall the perf testing framework needs a Sync Stack style change where synchronous tests are fully synchronous [at least within the code bounds of perf-test-core])